src/tedpy/ted6000.py

```diff
@@ -89,70 +81,50 @@ def polling_delay(self) -> int:
         """Return the delay between successive polls of MTU data."""
         return self.endpoint_settings_results["SystemSettings"]["MTUPollingDelay"]
 
+    def energy(self) -> EnergyYield:
+        """Return energy yield information for the whole system."""
+        data = self.endpoint_dash_results[0]["DashData"]
+        return EnergyYield(int(data["Now"]), int(data["TDY"]), int(data["MTD"]))
+
     @property
     def system_type(self) -> SystemType:
         """Return the MTU configuration of the system."""
-        return SystemType(
-            int(
-                self.endpoint_settings_results["SystemSettings"]["Configuration"][
-                    "SystemType"
-                ]
-            )
-        )
+        settings = self.endpoint_settings_results["SystemSettings"]
+        return SystemType(int(settings["Configuration"]["SystemType"]))
 
-    def total_consumption(self) -> EnergyYield:
-        """Return consumption information for the whole system."""
-        data = self.endpoint_dashboard_results[0]["DashData"]
-        return EnergyYield(
-            YieldType.SYSTEM_NET, int(data["Now"]), int(data["TDY"]), int(data["MTD"])
-        )
+    def _mtu_energy(self, mtu: TedMtu) -> EnergyYield:
+        """Return consumption or production information for a MTU."""
+        data = self.endpoint_mtudash_results[mtu.position]["DashData"]
+        return EnergyYield(int(data["Now"]), int(data["TDY"]), int(data["MTD"]))
 
-    def total_load(self) -> EnergyYield:
+    def consumption(self) -> EnergyYield:
         """Return load information for the whole system."""
-        data = self.endpoint_solardash_results[1]["DashData"]
-        return EnergyYield(
-            YieldType.SYSTEM_LOAD, int(data["Now"]), int(data["TDY"]), int(data["MTD"])
-        )
+        if self.system_type == SystemType.NET:
+            return self.energy()  # For NET type, consumption and energy are equal
+        data = self.endpoint_dash_results[1]["DashData"]
+        return EnergyYield(int(data["Now"]), int(data["TDY"]), int(data["MTD"]))
 
-    def total_generation(self) -> EnergyYield:
+    def production(self) -> EnergyYield:
         """Return generation information for the whole system."""
-        data = self.endpoint_solardash_results[2]["DashData"]
-        return EnergyYield(
-            YieldType.SYSTEM_GENERATION,
-            int(data["Now"]),
-            int(data["TDY"]),
-            int(data["MTD"]),
-        )
+        data = self.endpoint_dash_results[2]["DashData"]
+        return EnergyYield(-int(data["Now"]), -int(data["TDY"]), -int(data["MTD"]))
```

Comment:
From previous comments I assume you are expecting production to return a positive value? The instructions state that the production returned by TED should be negative when properly configured. When I was doing my testing above my system was returning positive values so it was not properly configured for production.

Reply:
I see from the unit tests that production is expected to return positive and the MTU itself is going to return negative.

Reply:
Yes; the MTU should return negative and the production() method should return positive.

Now I realize I should probably do a check on the MTU type and make sure MTUs configured as GEN return positive as well.